core/src/main/scala/filodb.core/downsample/DownsamplePeriodMarker.scala

```diff
@@ -0,0 +1,148 @@
+package filodb.core.downsample
+
+import debox.Buffer
+import enumeratum.{Enum, EnumEntry}
+import scalaxy.loops._
+
+import filodb.core.metadata.DataSchema
+import filodb.core.store.{ChunkSetInfoReader, ReadablePartition}
+import filodb.memory.format.PrimitiveVectorReader
+import filodb.memory.format.vectors.{CorrectingDoubleVectorReader, DoubleVectorDataReader}
+
+/**
+  * Enum of supported downsample period marker names
+  * @param entryName name of the marker
+  * @param periodMarkerClass its corresponding marker class used for instance construction
+  */
+sealed abstract class PeriodMarkerName(override val entryName: String, val periodMarkerClass: Class[_])
+  extends EnumEntry
+
+object PeriodMarkerName extends Enum[PeriodMarkerName] {
+  val values = findValues
+  case object Time extends PeriodMarkerName("time", classOf[TimeDownsamplePeriodMarker])
+  case object Counter extends PeriodMarkerName("counter", classOf[CounterDownsamplePeriodMarker])
+}
+
+/**
+  * Implementations of the trait identify row numbers that
+  * demark the boundaries of downsampling periods
+  */
+trait DownsamplePeriodMarker {
+
+  def name: PeriodMarkerName
+
+  /**
+    * Ids of Data Columns the marker works on.
+    * The column id values are fed in via downsampling configuration of the dataset
+    */
+  def inputColId: Int
```

Comment:
Comment says this reports Ids (plural) but definition only supports one Id.  Make this return Buffer[Int]?

Reply:
Only one column in current implementation. I dont want to over-engineer. Easy to change when there is a real use case (multiple counters which can drop at different times for example).